### ISSUE TITLE
[BUG]: Audio transcoding not working 

### ISSUE THREAD

**Issue by zenzo22** [tvheadend/tvheadend#1663]
### Tvheadend Version

v4.3 (development)

### Build Number

4.3-2286~gae51d24fe

### Platform

Ubuntu 20.04 (Focal)

### Architecture

ARMv8 (arm)

### Category

Transcoding

### Description

Hello,

asked this before without getting a solution: I am currently using HTS Tvheadend 4.3-2286~gae51d24fe on several root servers.
Due to limited mobile data volume I want to transcode video and audio to a lower bitrate.

Video transcoding is working fine but currently I can only use "copy" for the audio streams.
No matter which codec profiles I want to use for the audio transcoding to AAC 56k (or any other) no sound seems to be transcoded or at least output by TVH. 

The audio stream in VLC is always empty, using all kinds of different settings. Tried this for different networks and on different servers. Switching back to "copy" the original audio is working. FFProbe and the stream output on VLC show that there is no audio stream sent at all when transcoded.

It affects pretty much all recent versions (stable, development, unstable) of tvheadend.

I know this worked before (years ago though) and I have no clue what causes this behaviour. I have no stream filters applied. All neccessary codecs are installed and working. Manually transcoding the same input streams via the command line produces the desired audio output.

Would be really thankful for any help on this.

### Logfile

mpegts: channels - ARTE HD - tuning on IPTV #1
subscription: 0007: "HTTP" subscribing on channel "ARTE HD", weight: 100, adapter: "IPTV #1", network: "xxx", mux: "channels - ARTE HD", provider: "FFmpeg", service: "ARTE HD", profile="mobi", hostname="xxx", username="xxx", client="VLC/3.0.18 LibVLC/3.0.18"
transcode: 0005: 02:H264: ==> Using profile 400
transcode: 0005: 01:AAC: ==> Using profile 56k
libav: AVCodecContext: using SAR=1/1
libav: AVCodecContext: using cpu capabilities: ARMv8 NEON
libav: AVCodecContext: profile High, level 3.0, 4:2:0, 8-bit

### Reproduction

Tested on several devices with different linux distros and different TVH versions, different sources -> no transcoded audio output.

**Comment by Flole998** [comment]
You can try bisecting the issue, if it worked previously there is some commit that will break it. Once that is identified a solution can be worked on.

**Comment by Flole998** [comment]
Older versions are not available on Cloudsmith, you will have to self-compile older versions.

**Comment by Flole998** [comment]
Now you are comparing 2 completely different branches. You need to go back on the 4.3 branch to figure out where the issue was introduced there.

**Comment by GeekSpeak79** [comment]
I'll investigate further, but it appears that the transcoding has never worked in the 4.3 branch.

**Comment by Flole998** [comment]
IIRC it did work a few years ago. But there have been many library updates in the meantime, so it probably broke somewhere along the way.

**Comment by GeekSpeak79** [comment]
Good to know thanks. 
I'll see if i can identify a suspect commit. I traced it as far back to 6bfeca6c-ls13 on 4.3 branch where it still fails.

**Comment by Flole998** [comment]
Thats quite old, I'd expect it to work on that old releases. You are compiling with static ffmpeg, right? The dynamic versions that use your system ffmpeg might fail if it is a ffmpeg bug.

**Comment by GeekSpeak79** [comment]
Based on your recommendation I recompiled version 4.3-1906~g7e1dac822 ensuring FFMPEG was set to static. Transcode muxing issue persisted, however I did manage to get an error during the test that i have attached.
![error code](https://github.com/tvheadend/tvheadend/assets/168280536/e23d8e35-9721-41e3-833d-2563ff3fd609)

![TVHE options](https://github.com/tvheadend/tvheadend/assets/168280536/6f8b7f9b-3fb0-465d-9836-9a09e418b7a4)

**Comment by mrblond18** [comment]
Hello,
I took the leap and went from 4.2 to 4.3 (Build: 4.3-2323~g73a6bd00d) this week. I compiled myself on Ubuntu 22.04 64 bit. I am having this same issue. However, I have been able to figure out it only fails for aac source streams. If the source is AC3 it does properly transcode, also to aac.

During compiling I noticed:
```
libfdkaac                                no
libfdkaac_static                         no
```
I built with:
`$ AUTOBUILD_CONFIGURE_EXTRA=--enable-libffmpeg_static\ --enable-trace\ --enable-debug\ --enable-nonfree\ --enable-libfdk-aac\ --enable-libfdkaac_static ./Autobuild.sh`

I already had installed libfdk-aac0, libfdk-aac2 and libfdk-aac-dev on my system.

Any ideas how to fix?

**Comment by GeekSpeak79** [comment]
I've tried compiling commit versions back to 2017 and I haven't found a version of 4.3 Build:XXX that has a working transcode. If anyone knows a version of 4.3 that does work, I'm happy to trace the commits to identify where it stopped working.

Currently I just use 4.2.8-22~g67e75a932 for transcoding needs.

Docker builds have the same issue

**Comment by uknunknown** [comment]
Can you check if branch:
https://github.com/uknunknown/tvheadend/tree/dev_fix_aac_decode
is fixing your problem?

**Comment by mrblond18** [comment]
> Can you check if branch: https://github.com/uknunknown/tvheadend/tree/dev_fix_aac_decode is fixing your problem?

I compiled, installed and tested this version and it works! Thanks a lot.
Can you let me know once it is merged into main TVH repo?

I used this channel:
2024-08-27 12:17:52.645 mpegts: MTV 80s mux in ABC- tuning on IPTV #1
2024-08-27 12:17:52.667 subscription: 0001: "HTTP" subscribing on channel "MTV 80s", weight: 100, adapter: "IPTV #1", network: "ABC", mux: "MTV 80s mux", provider: "FFmpeg", service: "Service01", profile="TranscodeHD-600MBph", hostname="192.168.1.93", username="ukkepuk", client="TvhClient/933 LibVLC/3.0.19"
2024-08-27 12:17:52.733 transcode: 0001: 01:H264: ==> Using profile libx265-HD
2024-08-27 12:17:52.734 transcode: 0001: 02:AAC: ==> Using profile webtv-aac-HD

With my previous build:
Build: 4.3-2323-g73a6bd00d-dirty (2024-05-08T10:00:55+0000)
transcoding does not give audio.

With your build:
Build: 4.3-2358-g2b90c5e04 (2024-08-27T09:42:02+0000)
transcoding works without issues, also giving audio.

btw I am using this audio profile:
![image](https://github.com/user-attachments/assets/1a1c899a-1415-4ed1-bc5b-bc700999c359)

**Comment by GeekSpeak79** [comment]
Thank you for your efforts, greatly appreciated.

Am I correct in looking at the code: option 1 is more of a brute force patch guaranteeing aac will be tagged, and Option 2 is more of a try-catch solution looking for the input_GH?

If so does Option2 have a susceptibility to mis-identifying no input_gh? 

I hope I’m posing a properly organized question.

**Comment by uknunknown** [comment]
For some unknown reason, in previous implementation, not having input_gh is assuming a failure. I don't think this is true for audio. As for video ... the transcoding will fail later if is missing important information.

**Comment by Flole998** [comment]
Option 1 does apparently skip calling that function completely, option 2 calls it but then immediately exits because input_gh is not available. So why call it in the first place? I think option 1 is the better solution

**Comment by uknunknown** [comment]
The only advantage of option 2 is: assuming there is a satellite/IPTV stream AAC with input_gh populated (that probably was working also today), it is going to behave like before (not stripping that out). Option 1 on the other hand will strip down that info. 
Also this function is called also for h264; assuming there is a stream without input_gh that was not working before (because the function will return -11) it might be decoded after the fix (worst case will fail later ... down the path). 
My personal observation is that all audio stream that I found (not many on the other hand) don't have this information; and all video streams have this information.
Assuming you still prefer to go ahead with option 1: do you want to delete 'open' from all audio decoders? or only from AAC? I assume future IPTV might start using other codecs ... that are without royalties (like opus). It is only a meter of time until we will have to remove it from other audio codecs also.

**Comment by uknunknown** [comment]
@Flole998 Please restart the failed tests ... I don't think failures are related to the changes: https://github.com/tvheadend/tvheadend/pull/1763
